sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowCreateTableSuiteBase.scala

```diff
@@ -0,0 +1,145 @@
+    }
+
+    withGlobalTempView(viewName) {
+      sql(s"CREATE GLOBAL TEMPORARY VIEW $viewName AS SELECT 1 AS a")
+      val globalTempViewDb = spark.sessionState.catalog.globalTempViewManager.database
+      val ex = intercept[AnalysisException] {
+        sql(s"SHOW CREATE TABLE $globalTempViewDb.$viewName")
+      }
+      assert(ex.getMessage.contains(
+        s"$globalTempViewDb.$viewName is a temp view. " +
+          "'SHOW CREATE TABLE' expects a table or permanent view."))
+    }
+  }
+
+  test("SPARK-36012: ADD NULL FLAG WHEN SHOW CREATE TABLE") {
+    val db = "ns1"
+    val table = "tbl"
+    withNamespaceAndTable(db, table) { t =>
+      sql(
+        s"""
+           |CREATE TABLE $t (
+           |  a bigint NOT NULL,
+           |  b bigint
+           |)
+           |USING ${classOf[SimpleInsertSource].getName}
+        """.stripMargin)
+      val showDDL = getShowCreateDDL(s"SHOW CREATE TABLE $t")
+      // if v2 showDDL(0) == s"CREATE TABLE $t ("
+      // if v1 showDDL(0) == s"CREATE TABLE `$db`.`$table` ("
+      assert(showDDL(1) == "`a` BIGINT NOT NULL,")
+      assert(showDDL(2) == "`b` BIGINT)")
+      assert(showDDL(3) == s"USING ${classOf[SimpleInsertSource].getName}")
+    }
+  }
+
+  test("SPARK-24911: KEEP QUOTES FOR NESTED FIELDS") {
+    val db = "ns1"
+    val table = "tbl"
+    withNamespaceAndTable(db, table) { t =>
+      sql(s"CREATE TABLE $t (`a` STRUCT<`b`: STRING>) $defaultUsing")
+      val showDDL = getShowCreateDDL(s"SHOW CREATE TABLE $t")
+      // if v2 showDDL(0) == s"CREATE TABLE $t ("
+      // if v1 showDDL(0) == s"CREATE TABLE `$db`.`$table` ("
+      assert(showDDL(1) == "`a` STRUCT<`b`: STRING>)")
+      if (catalogVersion == "Hive V1") {
+        assert(showDDL(2) == "USING text")
+      } else {
+        assert(showDDL(2) == defaultUsing)
+      }
+    }
+  }
+
+  test("DATA SOURCE TABLE WITH USER SPECIFIED SCHEMA") {
+    val db = "ns1"
+    val table = "ddl_test"
+    withNamespaceAndTable(db, table) { t =>
+      val jsonFilePath = Utils.getSparkClassLoader.getResource("sample.json").getFile
+      sql(
+        s"""CREATE TABLE $t (
+           |  a STRING,
+           |  b STRING,
+           |  `extra col` ARRAY<INT>,
+           |  `<another>` STRUCT<x: INT, y: ARRAY<BOOLEAN>>
+           |)
+           |USING json
+           |OPTIONS (
+           | PATH '$jsonFilePath'
+           |)
+         """.stripMargin
+      )
+      val showDDL = getShowCreateDDL(s"SHOW CREATE TABLE $t")
+      // if v2 showDDL(0) == s"CREATE TABLE $t ("
+      // if v1 showDDL(0) == s"CREATE TABLE `$db`.`$table` ("
+      assert(showDDL(1) == "`a` STRING,")
+      assert(showDDL(2) == "`b` STRING,")
+      assert(showDDL(3) == "`extra col` ARRAY<INT>,")
+      assert(showDDL(4) == "`<another>` STRUCT<`x`: INT, `y`: ARRAY<BOOLEAN>>)")
+      assert(showDDL(5) == "USING json")
+      // V2 showDDL(6) == LOCATION 'jsonFilePath'
+      // V1 showDDL(6) == LOCATION 'file:jsonFilePath'
```

Comment:
Can you write a function which accept one expected string, splits it to lines and compares to the command output?